service/azureconfig/v1/resource/vnetpeering/delete.go

```diff
@@ -0,0 +1,66 @@
+package vnetpeering
+
+import (
+	"context"
+	// "fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/azure-operator/service/azureconfig/v1/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+func (r Resource) NewDeletePatch(ctx context.Context, azureConfig, current, desired interface{}) (*framework.Patch, error) {
+	a, err := key.ToCustomObject(azureConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewDeletePatch")
+	}
+	c, err := toVnetPeering(current)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewDeletePatch")
+	}
+	d, err := toVnetPeering(desired)
+	if err != nil {
+		return nil, microerror.Maskf(err, "NewDeletePatch")
+	}
+
+	return r.newDeletePatch(ctx, a, c, d)
+}
+
+func (r Resource) newDeletePatch(ctx context.Context, azureConfig providerv1alpha1.AzureConfig, current, desired network.VirtualNetworkPeering) (*framework.Patch, error) {
+	patch := framework.NewPatch()
+	patch.SetDeleteChange(current)
```

Comment:
Why always only deleting current and not desired?

Reply:
Deleting current ensure there is something to be deleted, since it has te be retrieved from azure first. Deleting desired does not make sense to me since it is always there.

I should probably also check for emptiness of current here, in case it is already gone.

Reply:
I see. My thinking is often that `GetCurrentState` has higher chances of being wrong for several reasons whereas `GetDesiredState` is often more static and not that error prone in comparison. So my thinking is usually considering the desired state to be deleted and be fine in case it is not found at all, which we have to deal with the other way around anyway.

Reply:
Makes actual sense, I will update that to desired and handle the not found case later on in `ApplyDeleteChange`